rotkehlchen/tests/api/test_addressbook.py

```diff
@@ -86,14 +86,72 @@ def test_get_addressbook(
         ),
         json={
             'addresses': [
-                {'address': generated_entries[1].address, 'blockchain': 'optimism'},
+                {
+                    'address': generated_entries[1].address,
+                    'blockchain': SupportedBlockchain.OPTIMISM.serialize(),
+                },
             ],
         },
     )
     result = assert_proper_response_with_result(response=response)
-    expected_entries = [generated_entries[1]]
-    deserialized_entries = [AddressbookEntry.deserialize(data=raw_entry) for raw_entry in result]
-    assert set(deserialized_entries) == set(expected_entries)
+    deserialized_entries = [AddressbookEntry.deserialize(data=raw_entry) for raw_entry in result['entries']]  # noqa: E501
+    assert set(deserialized_entries) == {generated_entries[1]}
+
+    # filter by name and blockchain
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'name_substring': 'neighbour',
+            'blockchain': SupportedBlockchain.ETHEREUM.serialize(),
+        },
+    )
+    result = assert_proper_response_with_result(response=response)
+    deserialized_entries = [AddressbookEntry.deserialize(data=raw_entry) for raw_entry in result['entries']]  # noqa: E501
+    assert set(deserialized_entries) == {generated_entries[2]}
+    assert result['entries_found'] == 1
+    assert result['entries_total'] == 5
+
+    # test api pagination
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'offset': 0,
+            'limit': 1,
+            'blockchain': SupportedBlockchain.ETHEREUM.serialize(),
+        },
+    )
+    result = assert_proper_response_with_result(response=response)
+    deserialized_entries = [AddressbookEntry.deserialize(data=raw_entry) for raw_entry in result['entries']]  # noqa: E501
+    assert set(deserialized_entries) == {generated_entries[0]}
+    assert result['entries_found'] == 2
+    assert result['entries_total'] == 5
+
+    # pagination offset works
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'offset': 1,
+            'limit': 1,
+            'blockchain': SupportedBlockchain.ETHEREUM.serialize(),
+        },
+    )
+    result = assert_proper_response_with_result(response=response)
+    deserialized_entries = [AddressbookEntry.deserialize(data=raw_entry) for raw_entry in result['entries']]  # noqa: E501
+    assert set(deserialized_entries) == {generated_entries[2]}
+    assert result['entries_found'] == 2
+    assert result['entries_total'] == 5
```

Comment:
After this you should also add a filter by a list of addresses. Make sure to have at least two tests. One for 1 address and 1 for 2 addresses.